### PR TITLE
Add code of conduct and contribution guidelines for maintainers

### PR DESCRIPTION
**Pull request by cschreib**
This PR introduces changes to the repository governance, to allow for more than one maintainer. See https://github.com/snitch-org/snitch/issues/132 for the related discussion. I am also taking the opportunity to consolidate some of the documentation.

Feedback welcome.

### CODE REVIEW

File: doc/testing_snitch.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        - [Compile-time unit tests](#compile-time-unit-tests-1)
+        - [Approval tests](#approval-tests-1)
+        - [Install tests](#install-tests-1)
+    - [Automated testing pipeline](#automated-testing-pipeline)
+
+<!-- /MarkdownTOC -->
+
+## Why
+
+If you are trying to [contribute](../CONTRIBUTING.md) to _snitch_, you should run the _snitch_ test suite to ensure your changes have not accidentally broken any existing feature. You should also ensure that any new feature you add is properly tested, so others don't break it with their changes later on.
```
Comment (tocic) [inline doc/testing_snitch.md:24]: I think it's more robust to use an "absolute" path here - `/CONTRIBUTING.md`. The same goes for `../.github/workflows`. See https://github.com/snitch-org/snitch/commit/9caaeec74611b626b0901e9af3c16ee4e4aadb44.

File: doc/testing_snitch.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+ - [compile-time unit tests](#compile-time-unit-tests) (not implemented yet)
+ - [approval tests](#approval-tests) (in `tests/approval_tests`)
+ - [install tests](#install-tests) (in `test/install_tests`)
+
+
+### Run-time unit tests
+
+The purpose of these tests is to check the detailed run-time behavior of _snitch_. This includes testing all foreseeable edge cases, happy path and sad path, etc. These tests contribute to the test coverage analysis.
+
+The run-time unit tests are roughly organized by theme into separate files in the `tests/runtime_tests` directory. New files can be added there if necessary. There is no enforced mapping between test files and library source files. Likewise, there is no enforced mapping between test case names and tested classes / functions; this is left of the best judgment of the contributor.
```
Comment (tocic) [inline doc/testing_snitch.md:40]:
```suggestion
The run-time unit tests are roughly organized by theme into separate files in the `tests/runtime_tests` directory. New files can be added there if necessary. There is no enforced mapping between test files and library source files. Likewise, there is no enforced mapping between test case names and tested classes / functions; this is left to the best judgment of the contributor.
```

File: doc/testing_snitch.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+Run-time unit tests come in two flavor: using _doctest_ to test _snitch_, or using _snitch_ to test itself. The latter is less safe, since a bug in _snitch_ might accidentally make these tests pass (e.g., if the test cases are not running at all), but it usually provides nicer tests results on failure.
```
Comment (tocic) [inline doc/testing_snitch.md:42]:
```suggestion
Run-time unit tests come in two flavors: using _doctest_ to test _snitch_, or using _snitch_ to test itself. The latter is less safe, since a bug in _snitch_ might accidentally make these tests pass (e.g., if the test cases are not running at all), but it usually provides nicer test results on failure.
```

File: doc/testing_snitch.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+
+### Compile-time unit tests
+
+Not implemented yet. The purpose of these tests will be to check how _snitch_ behaves at compile-time. In particular, to ensure that some specific usage will *not* compile. This cannot be achieved with the regular run-time tests, because it would prevent the whole test suite from compiling at all. Compile-time tests are better implemented at the build-system level, since the build system knows what compiler to use, what options, etc.
+
+
+### Approval tests
+
+These tests provide a more manageable way to check the output of _snitch_ for each reporter. Rather than use hand-written checks ("this line must contain this string, and then this other string"), which would be painful to write to achieve 100% coverage, the approach is to capture the output and save it to a file. If the output is satisfactory, we commit it as the new "expected" output, and future test runs just need to check that the output is compatible with the "expected" output. These tests contribute to the test coverage analysis.
```
Comment (tocic) [inline doc/testing_snitch.md:72]:
```suggestion
These tests provide a more manageable way to check the output of _snitch_ for each reporter. Rather than use handwritten checks ("this line must contain this string, and then this other string"), which would be painful to write to achieve 100% coverage, the approach is to capture the output and save it to a file. If the output is satisfactory, we commit it as the new "expected" output, and future test runs just need to check that the output is compatible with the "expected" output. These tests contribute to the test coverage analysis.
```
See https://github.com/snitch-org/snitch/pull/81#pullrequestreview-1380810467.

File: doc/testing_snitch.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+#### Install tests
+
+These are a bit cumbersome to run locally, so the recommendation is to let the automated testing pipeline run them for you.
+
+
+### Automated testing pipeline
+
+In case this document is out-of-date or you have any issue running the tests on your platform, please refer to the [CI scripts](../.github/workflows). The commands in these scripts will always be up-to-date, and you can execute them locally to run the tests on your own machine. Otherwise, GitHub will run these CI scripts once you open your PR, and then for each following commit you push to the branch once the PR is open. This consumes some resources, so please be considerate and try to push commits in bulk, once you think the code should pass tests, rather than one-by-one (i.e., avoid the "commit+push" workflow).
+
+The automated testing pipeline will run *all* tests, for *all* supported platforms. It will also includes other checks, such as test coverage checks. It is necessary for this pipeline to pass before a PR can be merged.
```
Comment (tocic) [inline doc/testing_snitch.md:144]:
```suggestion
The automated testing pipeline will run *all* tests, for *all* supported platforms. It will also include other checks, such as test coverage checks. It is necessary for this pipeline to pass before a PR can be merged.
```

**Review by tocic — APPROVED** [review]

File: doc/testing_snitch.md (hunk 1 above)
Comment (cschreib) [reply doc/testing_snitch.md]: Yep, done

File: README.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # snitch
 
-![Build Status](https://github.com/cschreib/snitch/actions/workflows/cmake.yml/badge.svg) [![codecov](https://codecov.io/gh/cschreib/snitch/branch/main/graph/badge.svg?token=X422DE81PN)](https://codecov.io/gh/cschreib/snitch)
+![Build Status](https://github.com/snitch-org/snitch/actions/workflows/cmake.yml/badge.svg) [![codecov](https://codecov.io/gh/snitch-org/snitch/branch/main/graph/badge.svg?token=X422DE81PN)](https://codecov.io/gh/snitch-org/snitch) [![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](CODE_OF_CONDUCT.md)
```
[the remark this answers is not in the document]
Comment (cschreib) [reply README.md:3]: They must have taken a bit of time to wake up; I did not do anything but the link works now.

[the remark this answers is not in the document]
Comment (tocic) [reply README.md:3]: Yes, it works for me too now.

**Review by tocic — APPROVED** [review]
